src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs

```diff
@@ -278,11 +278,11 @@ private void Intercept(IncomingMessage message, Action act)
                     this.pendingTip = chainedPendingTip; 
                 }
 
-                if ((newHeaders.Headers.Count != 0) && (pendingTipBefore.HashBlock != this.GetPendingTipOrChainTip().HashBlock))
+                if ((this.invalidHeaderReceived == false) && (newHeaders.Headers.Count != 0) && (pendingTipBefore.HashBlock != this.GetPendingTipOrChainTip().HashBlock))
```

Comment:
elsewhere we use !bool instead of bool == false

Reply:
done